src/Stream.ts

```diff
@@ -81,7 +81,7 @@ export const fromAsyncIterable = <A, R>(f: () => AsyncIterableWithReturn<A, R>):
   })
 )
 
-export const toAsyncIterable = <Error, Event, A>(fx: Fx.Fx<Async.Async | Fail.Fail<Error> | Stream<Event>, A>): AsyncIterableWithReturn<Event, A> => ({
+export const toAsyncIterable = <Errors extends Fail.Fail<any>, Events extends Stream<any>, A>(fx: Fx.Fx<Async.Async | Errors | Events, A>): AsyncIterableWithReturn<Event<Events>, A> => ({
```

Comment:
This works, too:

```ts
export const toAsyncIterable = <E extends Async.Async | Fail.Fail<any> | Stream<any>, A>(fx: Fx.Fx<E, A>): AsyncIterableWithReturn<Event<E>, A> => ({
```

I can't think of a compelling reason to prefer one over the other, though.